frontend/src/utils/useDebouncedEffect.js

```diff
@@ -0,0 +1,15 @@
+import { useCallback, useEffect } from "react";
+
+export const useDebouncedEffect = (effect, delay, deps) => {
```

Comment:
This is fire. I think we should copy what office hours does though and put this in a hooks folder. Like `frontend/src/hooks/useDebouncedEffect.js`